google/datalab/contrib/bigquery/commands/_bigquery.py

```diff
@@ -53,116 +53,179 @@ def _pipeline_cell(args, cell_body):
   transformation_config = bq_pipeline_config.get('transformation')
   output_config = bq_pipeline_config.get('output')
   parameters_config = bq_pipeline_config.get('parameters')
 
-  load_task_id = 'bq_pipeline_load_task'
-  execute_task_id = 'bq_pipeline_execute_task'
-  extract_task_id = 'bq_pipeline_extract_task'
-
-  load_task_config = _get_load_parameters(input_config)
-  execute_task_config = _get_execute_parameters(load_task_id, transformation_config,
-                                                output_config, parameters_config)
-  extract_task_config = _get_extract_parameters(execute_task_id, output_config)
-  pipeline_spec = {
-    'schedule': bq_pipeline_config['schedule'],
-  }
-
   pipeline_spec['tasks'] = {}
+
+  load_task_id = None
+  load_task_config = _get_load_parameters(input_config, transformation_config, output_config)
   if load_task_config:
+    load_task_id = 'bq_pipeline_load_task'
     pipeline_spec['tasks'][load_task_id] = load_task_config
+
+  execute_task_config = _get_execute_parameters(load_task_id, input_config, transformation_config,
+                                                output_config, parameters_config)
+  execute_task_id = None
   if execute_task_config:
+    execute_task_id = 'bq_pipeline_execute_task'
     pipeline_spec['tasks'][execute_task_id] = execute_task_config
+
+  extract_task_config = _get_extract_parameters(execute_task_id, input_config,
+                                                transformation_config, output_config)
   if extract_task_config:
-    pipeline_spec['tasks'][extract_task_id] = extract_task_config
-  pipeline_spec['emails'] = bq_pipeline_config.get('emails')
+    pipeline_spec['tasks']['bq_pipeline_extract_task'] = extract_task_config
 
   if not load_task_config and not execute_task_config and not extract_task_config:
     raise Exception('Pipeline has no tasks to execute.')
 
   return pipeline_spec
 
 
-def _get_load_parameters(bq_pipeline_input_config):
+def _get_load_parameters(bq_pipeline_input_config, bq_pipeline_transformation_config,
+                         bq_pipeline_output_config):
+    if bq_pipeline_input_config is None:
+      return None
+
     load_task_config = {'type': 'pydatalab.bq.load'}
 
-    # The path URL of the GCS load file(s).
+    # The path URL of the GCS load file(s), and associated parameters
     load_task_config['path'] = bq_pipeline_input_config.get('path')
 
+    if 'format' in bq_pipeline_input_config:
+      load_task_config['format'] = bq_pipeline_input_config['format']
+
+    if 'csv' in bq_pipeline_input_config:
+      load_task_config['csv_options'] = bq_pipeline_input_config['csv']
+
     # The destination bigquery table name for loading
-    load_task_config['table'] = bq_pipeline_input_config.get('table')
+    source_of_table = bq_pipeline_input_config
+    load_task_config['table'] = source_of_table.get('table')
+    if load_task_config['table'] is None and not bq_pipeline_transformation_config and \
```

Comment:
Our style guide says not to use line continuation slashes.  Wrap everything in parens instead.

Reply:
Done.